pkg/rbd/controllerserver.go

```diff
@@ -370,6 +345,9 @@ func (cs *ControllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 	rbdSnap.SourceVolumeID = req.GetSourceVolumeId()
 	rbdSnap.RequestName = req.GetName()
 
+	idLk := snapshotNameLocker.Lock(req.GetName())
```

Comment:
don't we need to take a lock on parent volume name? we may end up in race condition (volume delete and snap create)

Reply:
Yeah, as discussed in one of the previous PRs (iic stateless), there is a race scenario we could encounter.

Reply:
> don't we need to take a lock on parent volume name? we may end up in race condition (volume delete and snap create)

`genVolFromVolID` may pass or fail, if it fails (due to parent image not found or under deletion) the snapshot call fails immediately. If it passes the parent information is used to create the snapshot in `createSnapshot` call, and that passes the parent name to rbd for snapshot creation.

If the parent does not exist at this point in time, the snapshot will fail again, and required cleanup would be done.

If the delete on the parent is issued when a snap create is in flight to `rbd` that is something RBD would ensure to order and allow or disallow.

So there is no real reason for the plugin to lock and protect the parent from parallel calls, `rbd` will error out if the parent is missing, or handle a delete when a snap is in flight.

and tagging @dillaman to keep me honest around rbd statements made.